proposals/0298-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -1,108 +1,114 @@
 
-  1. Type of parameter is non-array enum (see example 1).
+  1. The type of parameter is a non-array enum (see example 1).
 
       SDL Core has to remove this parameter from the request since the value after cutting off becomes empty.
 
-  2. Parameter is an array of enum types (see example 2).
+  2. A parameter is an array of enum types (see example 2).
 
       SDL Core has to remove an unknown value from the array.
       A case is possible when all values are unknown and thus get removed.
       In this case, SDL Core has to proceed the same way as in case 1.
 
-  3. Parameter is part of the structure (see example 3).
+  3. A parameter is part of the structure (see example 3).
 
       SDL Core has to proceed the same way as in case #1 or #2 and remove this parameter from the structure.
       However, if the parameter is mandatory the structure becomes invalid.
       In this case, SDL Core has to remove the whole structure from the request.
       During this process, SDL Core has to proceed recursively from the very bottom level up to the top.
 
-  4. Parameter is a part of the structure which is a part of an array (see example 4).
+  4. A parameter is a part of the structure which is a part of an array (see example 4).
 
       SDL Core has to process it the same way as in case #3 and remove the structure as an item of the array.
       Once all the parameters are processed SDL Core has to proceed with the request as usual.
 
   If at least one value of at least one parameter was cut off, SDL Core has to update the response to the mobile application as follows:
 
   1. If the response was processed successfully, SDL Core has to provide the `WARNINGS` result code instead of `SUCCESS`.
-  2. SDL Core has to provide removed enum items in `info` string of the response.
+  2. SDL Core has to provide removed enum items in the `info` string of the response.
 
       Since there could be more than one parameter with cut-off value, `info` message can be constructed the following way:
 
       `Invalid enums were removed: <param_1>:<enum_value_1>,<enum_value_2>;<param_2>:<enum_value_3> ...`
 
-      If `info` parameter contains other value(s) belonging to the original processing result SDL Core has to append information about cut-off enum value(s) to the existing value.
+      If the `info` parameter contains other value(s) belonging to the original processing result SDL Core has to append information about cut-off enum value(s) to the existing value.
 
   **Examples:**
+
   In each example below the mobile application sends a request to SDL Core.
-  1. Request contains a parameter which value is an enum item.
+
+  **NOTE**
+
+  To determine the API version for choosing if a parameter is mandatory should be used the negotiated API version.
+
+  In the case when Mobile Application has the 7.0 API version and the SDL Core has the 6.0 API version, the negotiated API version will be 6.0 (the version of the SDL Core).
```

Comment:
```suggestion
  In the case when Mobile Application has the 7.0 Mobile API/RPC spec version and the SDL Core has the 6.0 Mobile API/RPC spec version, the negotiated version will be 6.0 (the version of the SDL Core).
```

Reply:
Updated in 3e4835a64afd1b1977009b77d4907a68f37120ee